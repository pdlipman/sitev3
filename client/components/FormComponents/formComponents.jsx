/* eslint-disable react/prop-types */ // --> OFF
import React from 'react';
import ReactMde, { ReactMdeCommands } from 'react-mde';

export const renderField = (
    {
        input,
        label,
        type,
        meta: {
            touched,
            error,
        },
    },
) => (
    <div>
        <input
            {...input}
            placeholder={label}
            type={type}
        />
        {touched && error && <span>{error}</span>}
    </div>
);


export const mdeEditorField = props => (
    <ReactMde
        value={props.input.value}
        onChange={param => props.input.onChange(param)}
        commands={ReactMdeCommands}
        textAreaProps={{
            id: 'ta1',
            name: 'ta1',
        }}
    />
);

/* eslint-enable react/prop-types */ // --> ON
