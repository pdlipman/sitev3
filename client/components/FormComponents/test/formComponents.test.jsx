import { shallow } from 'enzyme';
import { renderField } from '../formComponents';

describe('formComponent', () => {
    test('formComponent renders', () => {
        const label = 'Test Label';
        const type = 'text';
        const meta = {
            touched: false,
            error: '',
        };

        const testField = renderField({ label, type, meta });
        const wrapper = shallow(testField);
        expect(wrapper).toMatchSnapshot();
    });

    test('formComponent renders error', () => {
        const label = 'Test Label';
        const type = 'text';
        const meta = {
            touched: true,
            error: 'Error: Test Error',
        };

        const testField = renderField({ label, type, meta });
        const wrapper = shallow(testField);
        expect(wrapper).toMatchSnapshot();
    });
});
