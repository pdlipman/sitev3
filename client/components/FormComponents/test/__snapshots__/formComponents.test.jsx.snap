// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formComponent formComponent renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div>
    <input
      placeholder="Test Label"
      type="text"
    />
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          placeholder="Test Label"
          type="text"
        />,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "placeholder": "Test Label",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            placeholder="Test Label"
            type="text"
          />,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "placeholder": "Test Label",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`formComponent formComponent renders error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div>
    <input
      placeholder="Test Label"
      type="text"
    />
    <span>
      Error: Test Error
    </span>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          placeholder="Test Label"
          type="text"
        />,
        <span>
          Error: Test Error
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "placeholder": "Test Label",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Error: Test Error",
        },
        "ref": null,
        "rendered": "Error: Test Error",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            placeholder="Test Label"
            type="text"
          />,
          <span>
            Error: Test Error
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "placeholder": "Test Label",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Error: Test Error",
          },
          "ref": null,
          "rendered": "Error: Test Error",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
